Add resource labels to metrics

hlastras opened the pull request:
## Description
Add, by configuration, scanned resources labels into the prometheus metrics with a configurable prefix.

## Related issues
- Close #478

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/aquasecurity/trivy-operator?pullRequest=611) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/aquasecurity/trivy-operator?pullRequest=611) before we can accept your contribution.<br/><hr/>**hector.arranz** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/aquasecurity/trivy-operator?pullRequest=611) it.</sub>

hlastras commented:
The biggest changes have been in the `collector.go` file.

I had to do a little refactor to make it workable. I have moved all the prometheus descriptors, labels and severities that were in global variables, to a struct that will go inside the `ResourcesMetricsCollector`.
This is because the labels are no longer fixed, but instead vary according to the configuration, so they must be generated once we have the configuration data.
I have added a `constructor` for `ResourcesMetricsCollector` to build the labels and the descriptors based on the configuration.

I don't know if I have added enough tests, or if I should add any integration tests.
I haven't added any documentation yet until I'm sure the implementation will look like this.

I have tried to make small meaningful commits instead of one big one, to be able to see the evolution of the code. (even knowing that in the end there will be squashed).
First time using Conventional Commit.

And last but not least, yes, the names I have chosen are probably not the best, so I am totally open to changing them if there are suggestions.
